pkg/ebpf/events_definitions.go

```diff
@@ -43,7 +32,7 @@ type EventDefinition struct {
 	Name         string
 	Internal     bool
 	Syscall      bool
-	Probes       []probe
+	Probes       []probeDependency
```

Comment:
I actually had the same thought initially as well, but when I wrote it out it quickly got too nested for my taste, so I opted to keep it this way. 
And since almost all events besides the syscall ones have at least one probe, I think probes can get their own special category.

Reply:
So why not just keeping it "probe"?

Reply:
I still thought the name fits more, but it looked worse inside the Dependencies struct. I can rename it back if you find it important that it shouldn't be named xDependency if it's not inside Dependencies.

Reply:
We can keep it outside of dependencies, but I would keep it probe (no hard feelings about that though)